keras_core/ops/core.py

```diff
@@ -374,3 +375,35 @@ def convert_to_numpy(x):
         # We trigger it rather than duplicate it here.
         return np.array(x)
     return backend.convert_to_numpy(x)
+
+
+class Cond(Operation):
+    def __init__(self, true_fn, false_fn):
+        super().__init__()
+        self.true_fn = true_fn
+        self.false_fn = false_fn
+
+    def call(self, pred):
+        return backend.core.cond(pred, self.true_fn, self.false_fn)
+
+    def compute_output_spec(self, pred):
+        true_output_spec = backend.compute_output_spec(self.true_fn)
+        false_output_spec = backend.compute_output_spec(self.false_fn)
+        assert true_output_spec.dtype == false_output_spec.dtype
```

Comment:
Prefer using a `ValueError` that will print the different output shapes/dtypes -- that will be a better UX.